rust/src/delta.rs

```diff
@@ -540,8 +550,55 @@ impl DeltaTable {
             self.version(),
         );
 
-        while let PeekCommit::New(new_version, actions) =
-            self.peek_next_commit(self.version()).await?
+        let max_version = max_version.and_then(|x| {
+            if x <= self.version() {
```

Comment:
@rtyler I refactored this max_version handling a bit so it's clearer. Here I made explicit the behavior that's on `main` currently: i.e. the table will be updated to the latest version if max_version is smaller or equal to the current version.